java/ql/src/semmle/code/java/security/PathCreation.qll

```diff
@@ -0,0 +1,165 @@
+/**
+ * Models the different ways to create paths. Either by using `java.io.File`-related APIs or `java.nio.Path`-related APIs.
+ */
+
+import java
+import semmle.code.java.controlflow.Guards
+
+/** Models the creation of a path. */
+abstract class PathCreation extends Expr {
+  /** Gets an input that is used in the creation of this path. */
+  abstract Expr getInput();
```

Comment:
Should this be changed to `getAnInput`?
Maybe this should also clarify that this will only return inputs that are `String`s, if any?

Reply:
`getAnInput` would probably be better given that for example `Path.of(String, String...)` has multiple inputs.

> Maybe this should also clarify that this will only return inputs that are `String`s, if any?

This is currently not consistently the case, below `Paths.get` currently also matches `Paths.get(URI)`.
And in my opinion at least matching `URI` as well might be useful.

Reply:
I agree that `getAnInput` is a better name - let's change it.

Reply:
Regarding the qldoc clarification, perhaps add a line saying something like ``This excludes inputs of type `File` and `Path`.``